### ISSUE TITLE
Notification cannot be created in "finished" state

### ISSUE THREAD

**Issue by hypesystem** [argon/node-apn#240]
It would be nice if it was possible to configure the notification through an options object, instead of having to create it, and then set several attributes. Example of desired code (based on code [from the README](https://github.com/argon/node-apn#sending-a-notification)):

```javascript
var note = new apn.Notification({
    expiry: Math.floor(Date.now() / 1000) + 3600,
    badge: 3,
    sound: "ping.aiff",
    alert: "\uD83D\uDCE7 \u2709 You have a new message",
    payload: {'messageFrom': 'Caroline'}
});
```

**Comment by argon** [comment]
This has been addressed in #278. The new interface for `Notification` allows this:

```
> var note = new apn.Notification({
...     expiry: Math.floor(Date.now() / 1000) + 3600,
...     badge: 3,
...     sound: "ping.aiff",
...     alert: "\uD83D\uDCE7 \u2709 You have a new message",
...     payload: {'messageFrom': 'Caroline'}
... });
undefined
> note
{ encoding: 'utf8',
  payload: { messageFrom: 'Caroline' },
  compiled: false,
  aps:
   { badge: 3,
     sound: 'ping.aiff',
     alert: '📧 ✉ You have a new message' },
  expiry: 1462065249,
  priority: 10,
  truncateAtWordEnd: false }
> note.compile();
'{"messageFrom":"Caroline","aps":{"badge":3,"sound":"ping.aiff","alert":"📧 ✉ You have a new message"}}'
>
```

**Comment by hypesystem** [comment]
Nice!

**Comment by argon** [comment]
Fixed in v2.0.0
